Makefile

```diff
@@ -571,42 +597,70 @@ test: clean ## Run mlrun tests
 		$$IGNORE_ADDITION \
 		--forked \
 		-rf \
-		$$UNIT_TESTS_PATH
+		$$UNIT_TESTS_PATH ;\
+	if [ "$(COVERAGE)" = "true" ]; then \
+		echo "Unit test coverage report:"; \
+		COVERAGE_FILE=$$COVERAGE_FILE coverage report --rcfile=tests/tests.coveragerc; \
+	fi;
+
+
 
 .PHONY: test-integration-dockerized
 test-integration-dockerized: build-test ## Run mlrun integration tests in docker container
+	if [ "$(COVERAGE)" = "true" ]; then \
+		rm -rf /tmp/coverage_reports/integration_tests && mkdir -p /tmp/coverage_reports/integration_tests; \
+	fi; \
 	docker run \
 		-t \
 		--rm \
 		--network='host' \
 		-v /tmp:/tmp \
 		-v /var/run/docker.sock:/var/run/docker.sock \
-		$(MLRUN_TEST_IMAGE_NAME_TAGGED) make test-integration
+		-v /tmp/coverage_reports/integration_tests:/mlrun/tests/coverage_reports \
+		$(MLRUN_TEST_IMAGE_NAME_TAGGED) make test-integration COVERAGE=$(COVERAGE)
 
 .PHONY: test-integration
 test-integration: clean ## Run mlrun integration tests
-	python -m pytest -v \
+	set -e; \
+	if [ "$(COVERAGE)" = "true" ]; then \
+		rm -f tests/coverage_reports/integration_tests.coverage; \
+		export COVERAGE_FILE=tests/coverage_reports/integration_tests.coverage; \
+		COVERAGE_ADDITION="-m coverage run --rcfile=tests/tests.coveragerc"; \
+	else \
+		COVERAGE_ADDITION=""; \
+	fi; \
```

Comment:
this is duplicated a few times. suggesting something else:
Add inline makefile instead of part of the target block (need to resolve the coverage file name)
```
ifeq ($(COVERAGE),true)
	COVERAGE_FILE=tests/coverage_reports/integration_tests.coverage
	COVERAGE_ADDITION=-m coverage run --rcfile=tests/tests.coveragerc
	CLEAN_COVERAGE=rm -f tests/coverage_reports/integration_tests.coverage
else
	COVERAGE_FILE=
	COVERAGE_ADDITION=
	CLEAN_COVERAGE=true
endif
```
then in the unit/integration/migration target:
```
unit/integration/migration target:
	$(CLEAN_COVERAGE); \
	COVERAGE_FILE="$(COVERAGE_FILE)" \
	python $(COVERAGE_ADDITION) -m pytest tests/
```